mass_mailing_sending_queue/models/mail_mass_mailing.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from copy import deepcopy
+from openerp import api, fields, models, _
+from openerp.exceptions import Warning as UserError
+import logging
+
+
+class MailMassMailing(models.Model):
+    _inherit = 'mail.mass_mailing'
+
+    state = fields.Selection(selection_add=[
+        ('sending', "Sending"),
+    ])
+    pending = fields.Integer(string="Pending", compute='_compute_pending')
+    mass_mailing_sending_ids = fields.One2many(
+        comodel_name='mail.mass_mailing.sending', readonly=True,
+        inverse_name='mass_mailing_id', string="Sending tasks")
+
+    def read_group(self, cr, uid, domain, fields, groupby, **kwargs):
+        res = super(MailMassMailing, self).read_group(
+            cr, uid, domain, fields, groupby, **kwargs)
+        if groupby and groupby[0] == "state":
+            group_domain = domain + [('state', '=', 'sending')]
+            count = self.search_count(
+                cr, uid, group_domain, context=kwargs.get('context', {}))
+            res.append({
+                '__context': {'group_by': groupby[1:]},
+                '__domain': group_domain,
+                'state': ('sending', _("Sending")),
+                'state_count': count,
+            })
+        return res
+
+    def _sendings_get(self):
+        self.ensure_one()
+        return self.env['mail.mass_mailing.sending'].search([
+            ('mass_mailing_id', '=', self.id),
+            ('state', 'in', ('enqueued', 'sending')),
+        ])
+
+    @api.multi
+    def send_mail(self):
+        if self.env.context.get('sending_queue_enabled', False):
```

Comment:
Instead of adding one indentation and one level to the cyclomatic complexity, its' better to put:
```
if self.env.context.get('sending_queue_enabled'):
    return super(MailMassMailing, self).send_mail()
```
and repeat after the same super call